STIPS/STIP-005.md

```diff
@@ -0,0 +1,877 @@
+
+As the oracle price for a parent asset changes, the total margin requirement for an account scales with it such that the leverage ratio for a position maintains a constant relation to the position’s value in sport markets. (See  Appendix/Simulations/Leverage for an example).
+
+#### Funding
+
+Funding is a payment flow between makers and takers in the Perp system that acts as a balancing force to bring the AMM price of the virtual asset closer to the oracle price.
+
+A simplified formula for how these payments are calculated is:
+
+```
+   time = seconds since position opened / 86400
+   pendingFundingPayment = _positionSize * (AMM market price - Oracle price) *  time
+```
+
+If a taker opens a long position and the oracle price is below AMM market, they pay funding fees. If the oracle price is above AMM market price, they receive them. Funding payments are settled when the trader closes a position, realizing their PNL. They are calculated using the  AMM price and the 15 min TWAP oracle price read during settlement.
+
+Traders settle their pending funding payments whenever they open, modify or close a position. For the module's purposes this means funding is credited or debited during issuance, redemption, lever, and delever.
+
+#### Insurance Fund
+
+This is currently WIP with no unit tests in the perp-lushan repository, as of commit: 56ab296 (Sept 15)
+
+## Integration with Set Protocol / Feasibility Analysis
+
+**Fundamental Concepts**
+
+* Issuance replicates the current position (e.g its leverage and ratio of components).  USDC enters our system and we allocate resources to preserve the sets existing properties as we open positions on the external protocol.
+* Redemption is about replicating the current position as USDC leaves our system and we close positions on the external protocol.
+
+**Design Goals**
+
+* Works seamlessly with the existing DIM and SetToken
+* Works well with rest of the SetProtocol architecture and processes for managing Sets
+* Encapsulates all PerpV2 interactions in a dedicated module
+
+**Summary**
+
+We think PERP accounts should be modeled on the SetToken as an external USDC position and that PERP account virtual positions should be modeled as default positions on a virtual SetToken. This virtual SetToken is intended solely as a data structure which tracks position quantities within the PERP system. Using it lets us manage PERP accounts as a conventional SetToken component while preserving the ability to map PERP virtual positions onto SetProtocol’s existing logic for operations like replication and rebalancing.
```

Comment:
What's the purpose of the virtual SetToken? Feels like more state we have to manage/update. Does it help with issuance or redemption somehow?

Reply:
Yes. It serves as a way to track the different virtual assets held by SetToken and their respective amounts per SetToken.  We can't track the virtual assets in our "real" SetToken, as doing so would lead to changes bubbling up the stack. Virtual SetToken is a way to avoid those changes bubbling up the stack, while still be able to track virtual assets as just another regular ERC20 asset.

Reply:
It does help with issuance and redemption, as it essentially gives us the quantity of virtual assets held "per" SetToken. And fits quite well with our existing infrastructure of using "position units" as the base mathematical construct at various places in our stack.